KoPrivateGPT/retrieval/base.py

```diff
@@ -1,14 +1,21 @@
+import itertools
 from abc import ABC, abstractmethod
-from typing import List, Union
+from typing import List, Union, Dict
 from uuid import UUID
 
-from KoPrivateGPT.DB.base import BaseDB
+from KoPrivateGPT.DB import MongoDB, PickleDB
 from KoPrivateGPT.schema import Passage
+from KoPrivateGPT.utils.linker import RedisDBSingleton
 
 
 class BaseRetrieval(ABC):
+    def __init__(self):
+        self.check_duplicate = []
```

Comment:
self.check_duplicate can be List[DBOrigin] type.

Reply:
List[DBOrigin] -> List[BaseDB]